.ci/Jenkinsfile

```diff
@@ -34,13 +34,14 @@ pipeline {
         deleteDir()
         gitCheckout(basedir: "${BASE_DIR}")
         stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+        setEnvVar("GO_VERSION", readFile("${env.WORKSPACE}/${env.BASE_DIR}/.go-version").trim())
       }
     }
     stage('Check') {
       options { skipDefaultCheckout() }
       steps {
         cleanup()
-        withGoEnv(){
+        withGoEnv(version: "${GO_VERSION}"){
```

Comment:
`withGoEnv()` will search for the GO_VERSION or the file .go-version because it uses `goDefaultVersion()`

https://github.com/elastic/apm-pipeline-library/blob/master/vars/withGoEnvUnix.groovy#L37
https://github.com/elastic/apm-pipeline-library/blob/master/vars/goDefaultVersion.groovy#L19

Reply:
We must be missing something then, as previous executions are not setting the GO_VERSION properly: https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Ffleet-server/detail/PR-289/12/pipeline/143. As you can see there, the version is 1.15.8, even though the .go-version file for that PR is set to 1.16.2 (https://github.com/elastic/fleet-server/blob/60c9b4647d913e018471fb790629a27f65097786/.go-version)

Reply:
@kuisathaverat after reading the code in the pipeline step, I'm pretty sure this line https://github.com/elastic/apm-pipeline-library/blob/master/vars/goDefaultVersion.groovy#L29 is evaluated to `false`, therefore the `goDefaultVersion` is not overridden and it takes the default value: 1.15.8 (https://github.com/elastic/apm-pipeline-library/blob/master/vars/goDefaultVersion.groovy#L41)

Reply:
In this PR, we are reading the file from:
- this PR: `readFile("${env.WORKSPACE}/${env.BASE_DIR}/.go-version")`
- in the step: `readFile(file: ".go-version")?.trim()` 

So I think the step should always read from BASE_DIR, as it's a common pattern we use. Do you think we should open a bug in the library?

Reply:
we should run `withGoEnv()` at roo source level

Reply:
Ahh good catch! Will send a commit with that change, thanks!